codec-http/src/main/java/io/netty/handler/codec/http/websocketx/WebSocketProtocolHandler.java

```diff
@@ -96,8 +96,8 @@ public void close(final ChannelHandlerContext ctx, final ChannelPromise promise)
     }
 
     @Override
-    public void write(final ChannelHandlerContext ctx, Object msg, ChannelPromise promise) {
-        if (closeSent != null) {
+    public void write(final ChannelHandlerContext ctx, Object msg, Promise<Void> promise) {
+        if (closeSent != null && closeSent != promise) {
```

Comment:
There's some reentrancy that I'm not sure how was allowed to work before. Basically, when you close a web socket stream, it first assigns the closedSent promise, and writes the close frame, which gets us back into this method where we check if the stream has been closed. This normally prevents writes, but we can't allow ourselves to prevent the write of the close frame.